src/Microsoft.DotNet.Watcher.Tools/Program.cs

```diff
@@ -44,49 +65,39 @@ public static int Main(string[] args)
                     ev.Cancel = false;
                 };
 
-                return new Program().MainInternal(args, ctrlCTokenSource.Token);
+                return new Program(ctrlCTokenSource.Token, Console.Out, Console.Error).MainInternal(args);
             }
         }
 
-        private int MainInternal(string[] args, CancellationToken cancellationToken)
+        private int MainInternal(string[] args)
         {
-            var app = new CommandLineApplication();
-            app.Name = "dotnet-watch";
-            app.FullName = "Microsoft dotnet File Watcher";
-
-            app.HelpOption("-?|-h|--help");
+            var options = CommandLineOptions.Parse(args, _out);
+            if (options == null)
+            {
+                // invalid args syntax
+                return 1;
+            }
 
-            app.OnExecute(() =>
+            if (options.IsHelp)
             {
-                var projectToWatch = Path.Combine(Directory.GetCurrentDirectory(), ProjectModel.Project.FileName);
-                var watcher = DotNetWatcher.CreateDefault(_loggerFactory);
+                return 2;
+            }
 
-                try
-                {
-                    watcher.WatchAsync(projectToWatch, args, cancellationToken).Wait();
-                }
-                catch (AggregateException ex)
-                {
-                    if (ex.InnerExceptions.Count != 1 || !(ex.InnerException is TaskCanceledException))
-                    {
-                        throw;
-                    }
-                }
-
-                return 0;
-            });
-
-            if (args == null ||
-                args.Length == 0 ||
-                args[0].Equals("--help", StringComparison.OrdinalIgnoreCase) ||
-                args[0].Equals("-h", StringComparison.OrdinalIgnoreCase) ||
-                args[0].Equals("-?", StringComparison.OrdinalIgnoreCase))
+            var projectToWatch = Path.Combine(Directory.GetCurrentDirectory(), ProjectModel.Project.FileName);
+
+            try
             {
-                app.ShowHelp();
-                return 1;
+                DotNetWatcher
+                    .CreateDefault(_loggerFactory)
+                    .WatchAsync(projectToWatch, options.RemainingArguments, _cancellationToken)
+                    .Wait();
+            }
+            catch (AggregateException ex) when (ex.InnerExceptions.Count == 1 && ex.InnerException is TaskCanceledException)
```

Comment:
you could make MainInternal async and then catch only TaskCanceledException. You would `.Wait()` in Main instead of here.